packages/nx/src/lock-file/npm.ts

```diff
@@ -177,10 +197,17 @@ function mapPackageDependency(
 
     mappedPackages[packageName][key] = {
       ...(value as Omit<PackageDependency, 'packageMeta'>),
-      ...(!value.integrity && {
-        actualVersion: value.version,
-        version: value.resolved,
-      }),
+      ...(!value.integrity &&
+        value.version && {
```

Comment:
This `&& value.version &&` is the only addition. The rest of the file changes come from https://github.com/nrwl/nx/pull/13585